Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1

```diff
@@ -186,25 +149,19 @@ else
 
     $connection = Get-VssConnection -TaskContext $distributedTaskContext
 
-    Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName" -Verbose
-    $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -Connection $connection -TaskContext $distributedTaskContext
+    Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName with filter $machineFilter" -Verbose
+    $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -Connection $connection -TaskContext $distributedTaskContext -ResourceFilter $machineFilter
     Write-Verbose "Completed Register-Environment cmdlet call for environment : $environmentName" -Verbose
 
     $fetchedEnvironmentName = $environment.Name
 
-    if($resourceFilteringMethod -eq "tags")
-    {
-        $wellFormedTagsList = Get-WellFormedTagsList -tagsListString $machineFilter
+    Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $fetchedEnvironmentName" -Verbose
+    $resources = Get-EnvironmentResources -EnvironmentName $fetchedEnvironmentName -TaskContext $distributedTaskContext
+    Write-Verbose "Completed Get-EnvironmentResources cmdlet call for environment name: $fetchedEnvironmentName" -Verbose
 
-        Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $fetchedEnvironmentName with tag filter: $wellFormedTagsList" -Verbose
-        $resources = Get-EnvironmentResources -EnvironmentName $fetchedEnvironmentName -TagFilter $wellFormedTagsList -TaskContext $distributedTaskContext
-        Write-Verbose "Completed Get-EnvironmentResources cmdlet call for environment name: $fetchedEnvironmentName with tag filter" -Verbose
-    }
-    else
+    if ($resources.Count -eq 0)
     {
-        Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $fetchedEnvironmentName with machine filter: $machineFilter" -Verbose
-        $resources = Get-EnvironmentResources -EnvironmentName $fetchedEnvironmentName -ResourceFilter $machineFilter -TaskContext $distributedTaskContext
-        Write-Verbose "Completed Get-EnvironmentResources cmdlet call for environment name: $fetchedEnvironmentName with machine filter" -Verbose
+         throw (Get-LocalizedString -Key "No machine exists under environment: '{0}' for deployment" -ArgumentList $environmentName)
```

Comment:
User facing messages should not have 'environment' word. a) environment has a different meaning in RM. b) I am not sure if the notion of environment is meaningful when a flat list is given (will this message ever come for flatlist or variable?)